File: Dotnet/Loan/Class/HomeLoan.cs
namespace Class;
 public interface Discountable{
    double GetDiscount();
 } 
public class HomeLoan : Loan, Discountable
{

    public override float GetRate()
    {
        if (Principle<= 2000000)
        {
            return 0.10f; 
        }
        else
        {
            return 0.11f; 
        }
    }
    public double GetDiscount()
    {
      double result = Principle * GetRate() / 100;
      return result * 0.05; }
}

File: Java/Assignment/BankLoans/src/LoanType/HomeLoan.java
package LoanType;

interface Discountable
{
   public double GetDiscount();
}
public class HomeLoan implements Discountable
{
    int p;
    public HomeLoan(int p) {
      this.p=p;
    }

    public float GetRate()
    {
        if (p<= 20000)
        {
            return 0.10f;
        }
        else
        {
            return 0.11f;
        }
    }
    public double GetDiscount()
    {
        double result = p * GetRate() / 100;
        return result * 0.05;
    }
}
